tool/util/shared.go

```diff
@@ -9,8 +9,30 @@ import (
 
 const (
 	BuildTempDir = ".otel-build"
+	OtelRoot     = "github.com/open-telemetry/opentelemetry-go-compile-instrumentation"
 )
 
+// GetBuildTemp returns the path to the build temp directory $BUILD_TEMP/name
 func GetBuildTemp(name string) string {
 	return filepath.Join(BuildTempDir, name)
 }
+
+// BackupFile backups the source file to $BUILD_TEMP/backup/name, error is
+// tolerated as it's not critical.
+func BackupFile(names []string) {
```

Comment:
What about (just to avoid duplication
```suggestion
func copyFilesHelper(names []string, srcRoot, dstRoot string) {
	for _, name := range names {
		src := filepath.Join(srcRoot, name)
		dst := filepath.Join(dstRoot, name)
		_ = CopyFile(src, dst) // Error is tolerated
	}
}

func BackupFile(names []string) {
	backupDir := GetBuildTemp("backup")
	copyFilesHelper(names, "", backupDir)
}

func RestoreFile(names []string) {
	backupDir := GetBuildTemp("backup")
	copyFilesHelper(names, backupDir, "")
}
```